Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -285,9 +285,16 @@ public class PBXProjGenerator {
             }.flatMap { $0 }
         ).sorted()
 
-        var projectAttributes: [String: Any] = ["LastUpgradeCheck": project.xcodeVersion]
-            .merged(project.attributes)
-
+        var projectAttributes: [String: Any] = project.attributes
+        
+        // Set default LastUpgradeCheck if user did not specify
+        let lastUpgradeKey = "LastUpgradeCheck"
+        if !projectAttributes.contains(where: { (key, value) -> Bool in
+            key == lastUpgradeKey && value is String
+        }) {
+            projectAttributes[lastUpgradeKey] = project.xcodeVersion
```

Comment:
This code is the same as before, but it only replaces the key if it's a string, correct? Were you running into cases where devs were putting numbers in the project attributes?

Reply:
Almost. It will replace the key if the user did not specify a value for `LastUpgradeCheck` or if this value is not a string. In this case, it will keep the same behavior as today, using the `project.xcodeVersion` value.
I  just felt it would be safer to validade if the value we're inserting is a `String` to avoid any strange values.